src/i_flmusic.c

```diff
@@ -168,7 +169,8 @@ static boolean I_FL_InitMusic(void)
 static void I_FL_SetMusicVolume(int volume)
 {
     // FluidSynth's default is 0.2. Make 1.0 the maximum.
-    fluid_synth_set_gain(synth, (float)volume / 15);
+    // 0 -- 0.2 -- 10.0
+    fluid_synth_set_gain(synth, ((float)volume / 15) * (float)mus_gain);
```

Comment:
Isn't this a bit coarse? I mean the next step would be `mus_gain = 2` which would double up the volume and the final step would amplify by factor 10?

Reply:
How about this:
```diff
diff --git a/src/i_flmusic.c b/src/i_flmusic.c
index eb21843..6885b22 100644
--- a/src/i_flmusic.c
+++ b/src/i_flmusic.c
@@ -170,7 +170,7 @@ static void I_FL_SetMusicVolume(int volume)
 {
     // FluidSynth's default is 0.2. Make 1.0 the maximum.
     // 0 -- 0.2 -- 10.0
-    fluid_synth_set_gain(synth, ((float)volume / 15) * (float)mus_gain);
+    fluid_synth_set_gain(synth, ((float)volume / 15) * ((float)mus_gain / 10));
 }

 static void I_FL_PauseSong(void *handle)
diff --git a/src/m_misc.c b/src/m_misc.c
index bbc32f1..90a0cc9 100644
--- a/src/m_misc.c
+++ b/src/m_misc.c
@@ -2204,7 +2204,7 @@ default_t defaults[] = {
   {
     "mus_gain",
     (config_t *) &mus_gain, NULL,
-    {1}, {1, 10}, number, ss_none, wad_no,
+    {10}, {1, 100}, number, ss_none, wad_no,
     "amplify FluidSynth output level"
   },
 #endif
```

Reply:
Looks better, but let's be consistent and use percents like Pr+:
https://github.com/coelckers/prboom-plus/blob/d7f95371d59e6889c61f02aea65e2d28a9f2b2a5/prboom2/src/MUSIC/flplayer.c#L167

Reply:
So it will be 0-1000%? OK.

Reply:
👍

Reply:
Let's say 10-1000.